src/file/source-map/extract-source-map.js

```diff
@@ -1,16 +1,30 @@
 var convert = require('convert-source-map');
 
 var sourceMapRegex = /\n\/(\*|\/)# sourceMappingURL=(.*)$/mg;
+var inlineBase64SourceMapRegex = /\n\/(\*|\/)# sourceMappingURL=data:application\/json(.*);base64,(.*)$/mg;
+
+function extract(code, mapFileDir) {
+	
+	var inlineBase64SourceMapMatch = code.match(inlineBase64SourceMapRegex);
+	
+	if(inlineBase64SourceMapMatch) {
+		
+        return {
+            code: code.replace(sourceMapRegex, ''),
+            map: convert.fromComment(inlineBase64SourceMapMatch[0]).toObject()
+        };
+		
+	}
 
-function extract(code) {
-
-    var sourceMap = code.match(sourceMapRegex);
+    var sourceMapMatch = code.match(sourceMapRegex);
 
-    if (sourceMap) {
+    if (sourceMapMatch) {
 
+		var sourceMap = convert.fromMapFileComment(sourceMapMatch[0], mapFileDir);
+		
         return {
             code: code.replace(sourceMapRegex, ''),
-            map: convert.fromComment(sourceMap[0]).toObject()
+            map: sourceMap && sourceMap.toObject()

```

Comment:
Is this supposed to be `&&`? Won't this turn into a boolean?

Reply:
nope. this is kinda like the equivalent of `map = sourceMap?.sourceMap.toObject()` or `map = sourceMap == null ? null : sourceMap.toObject();` in C# land

![image](https://user-images.githubusercontent.com/8292698/53367686-ce03de80-3914-11e9-899f-470a9462683c.png)